tools/phenotype_annotation/phenotype_annotation

```diff
@@ -0,0 +1,337 @@
+      print_phenos($entry, \@extracted_phenos, $outType);
+    }
+  }
+  $parser->close();
+}
+
+sub pheno_phenotypeFeatures{
+  my ($inF) = @_;
+
+  my $ont_ad  = $registry->get_adaptor( 'Multi', 'Ontology', 'OntologyTerm');
+  my $parser = Bio::EnsEMBL::IO::ListBasedParser->open($inF);
+
+  while ($parser->next()){
+    my $entry = $parser->get_value();
+    my @extracted_phenos;
+    if ($entry =~ /(^[a-zA-Z]+:\d+)/ ){ # input is an Ontology Accession Term
+      @extracted_phenos = @{$pfa->fetch_all_by_phenotype_accession_type_source($entry, $mapping_type)};
+    } else {
+      my $terms = $ont_ad->fetch_all_by_name($entry);
+      if (scalar @$terms == 0){
+        warn "WARNING: phenotype description '$entry' not found in the ontology database\n" if (scalar @$terms == 0 && $verbose);
+        $terms = $pa->fetch_by_description_accession_type($entry, $mapping_type);
+        warn "WARNING: phenotype description '$entry' not found in the variation database\n" if (scalar @$terms == 0 && $verbose);
+        warn "WARNING: phenotype description '$entry' not found\n" if (scalar @$terms == 0);
+      }
+
+      foreach my $term (@$terms){
+        if (ref($term) eq 'Bio::EnsEMBL::OntologyTerm'){
+          my $acc = $term->accession;
+          my $current_phenos = $pfa->fetch_all_by_phenotype_accession_type_source($acc, $mapping_type);
+          warn "INFO: found ", scalar @$current_phenos, " assignedments of phenotype $acc\n" if $verbose;
+          push @extracted_phenos,@{$current_phenos};
+        } else {
+          foreach my $acc (@{$term->ontology_accessions}){
+            my $current_phenos = $pfa->fetch_all_by_phenotype_accession_type_source($acc, $mapping_type);
+            warn "INFO: found ", scalar @$current_phenos, " assignedments of phenotype $acc\n" if $verbose;
+            push @extracted_phenos,@{$current_phenos};
+          }
+        }
+      }
+    }
+    #make phenotype features unique
+    my %phenotypes = map { $_->phenotype_id. $_->seq_region_name. $_->seq_region_start.$_->seq_region_end.$_->object_id => $_ } @extracted_phenos;
+    @extracted_phenos = values %phenotypes;
+
+    print_phenos($entry, \@extracted_phenos, $outType);
+  }
+  $parser->close();
+}
+
+sub snp_phenotypeFeatures{
+  my $inF = shift;
+  my $fileType = shift;
+
+  my $fileTypeVcf= 0;
+  my $fileTypeVep = 0;
+  my $parser;
+  if (! defined $fileType){
+    $parser = Bio::EnsEMBL::IO::ListBasedParser->open($inF);
+  } elsif ($fileType eq 'vcf'){
+    $parser = Bio::EnsEMBL::IO::Parser::VCF4->open($inF);
+    $fileTypeVcf = 1;
+  } elsif ($fileType eq 'vep'){
+    $parser = Bio::EnsEMBL::IO::Parser::VEP_output->open($inF);
+    $fileTypeVep = 1;
+  } else {
+    throw IOException("Error opening file <$inF> - unknown file type <$fileType>");
+  }
+
+  my $coord_system = $ca->fetch_by_name('chromosome');
+
+  my $record;
+  while ($parser->next()){
+    if ($fileTypeVcf){
+      $record = $parser->{'record'};
+    } elsif ($fileTypeVep){
```

Comment:
If a variant has many overlapping transcripts, you get a lot of duplicates in the output file. You could save the name of the variant for which phenotype data is extracted and only process the next line if it is different to the last processed.